src/Aspire.Dashboard/Components/Dialogs/TextVisualizerDialog.razor.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Text.Json;
+using System.Xml;
+using System.Xml.Linq;
+using Aspire.Dashboard.Model.Otlp;
+using Microsoft.AspNetCore.Components;
+
+namespace Aspire.Dashboard.Components.Dialogs;
+
+public partial class TextVisualizerDialog : ComponentBase
+{
+    private const string XmlFormat = "xml";
+    private const string JsonFormat = "json";
+    private const string PlaintextFormat = "text";
+
+    private List<SelectViewModel<string>> _options = null!;
+    private string _selectedOption = null!;
+
+    private string _formattedText = string.Empty;
+
+    private readonly string _openSelectFormatButtonId = $"select-format-{Guid.NewGuid():N}";
+    private bool _isSelectFormatPopupOpen;
+
+    protected override void OnInitialized()
+    {
+        _options =
+        [
+            new SelectViewModel<string> { Id = PlaintextFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogPlaintextFormat)] },
+            new SelectViewModel<string> { Id = JsonFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogJsonFormat)] },
+            new SelectViewModel<string> { Id = XmlFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogXmlFormat)] },
+        ];
+    }
+
+    protected override void OnParametersSet()
+    {
+        // We don't know what format the string is in, but we can guess
+        if (TryFormatXml())
+        {
+            _selectedOption = XmlFormat;
+            return;
+        }
+
+        if (TryFormatJson())
+        {
+            _selectedOption = JsonFormat;
+            return;
+        }
+
+        _selectedOption = PlaintextFormat;
+        _formattedText = Content.Text;
+    }
+
+    private bool TryFormatXml()
+    {
+        try
+        {
+            _formattedText = XElement.Parse(Content.Text).ToString();
+            return true;
+        }
+        catch (XmlException)
+        {
+            // If the XML is invalid, just show the original text
+            _formattedText = Content.Text;
+            return false;
+        }
+    }
+
+    private bool TryFormatJson()
+    {
+        try
+        {
+            using var doc = JsonDocument.Parse(
+                Content.Text,
+                new JsonDocumentOptions
+                {
+                    AllowTrailingCommas = true,
+                    CommentHandling = JsonCommentHandling.Skip
```

Comment:
I don't think we want to be skipping comments!

Reply:
Our choices are either `Skip` or `Disallow`, we don't have the option to allow. Calling this method with `Allow` will always result in a JsonException, it is not allowed. It is preferable to `Skip` as opposed to `Disallow`

Reply:
`Note that, as of .NET Core 3.1, JsonDocument and JsonSerializer only support skipping or disallowing of comments; they do not support loading them. If you try to set JsonCommentHandling.Allow for either, you will get an exception:`

Reply:
That's a shame. I'd consider using `Disallow` then, so that no information is lost.

Alternatively, add an infobar to the UI that's shown when comments are removed. To detect this we could parse with `Disallow` then again with `Skip` if the first parse throws.